lib/replica/home.dart

```diff
@@ -53,16 +55,38 @@ class _ReplicaHomeScreenState extends State<ReplicaHomeScreen> {
 
   @override
   Widget build(BuildContext context) {
+    print("Home called");
     // We are showing the ReplicaSearchScreen here since we want the bottom tabs to be visible (they are not if it's its own route)
     // <08-23-22, kalli>  Not ideal UX - maybe add a spinner? Not sure
     // <09-07-22, kalli> Update after testing a debug build - this is not very noticeable.
     if (showResults) {
       return ReplicaSearchScreen(
         currentQuery: currentQuery,
         currentTab: currentTab,
+        onBackButtonPressed: onBackButtonPressed,
       );
     }
+    // we need to initialize controller again coz we when switch widget controller is getting disposed
+    _textEditingController = CustomTextEditingController(formKey: _formKey);
```

Comment:
This isn't a problem with your changes, but I noticed we aren't disposing this controller when the widget is no longer needed. Should we be doing that?

```dart
@override
void dispose() {
  _textEditingController.dispose();
}    
```

Reply:
Thanks for caching this, I will add this.